### ISSUE TITLE
compact display of barnstars on /profile/ pages

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#705]
We'd like to see if barnstars could be displayed in a more compact way on profile pages like https://publiclab.org/profile/warren or https://publiclab.org/profile/cfastie.

Barnstar display code is currently here: https://github.com/publiclab/plots2/blob/master/app/views/users/profile.html.erb#L144-L149

But keep in mind this is only for the smaller "list-style" barnstars -- the big ones with custom graphics are simply part of the profile text itself (from an older way of giving barnstars). We're only looking to make the smaller ones more compact for now. 

Two options are:

1. in a new tab, preferably with the count of barnstars (using `<%= @user.user.barnstars.length %>` to display the count) 
2. we could show 2-3 barnstars, and if there are more, we could show a link to expand the rest, labelled `4 more barnstars`. This would be slightly more complex to code, as you'd have to programmatically separate the first 3 from the remaining ones. 

Happy to help either way, so please ask!

**Comment by carolineh101** [comment]
Hello, I'd like to help out with this!

**Comment by ananyo2012** [comment]
@jywarren Before starting with this it would be great if you could re-upload the barnstar images in the wiki page with the similar image name as the title since I don't see similar image paths for the images. This will help in this automation.

**Comment by ananyo2012** [comment]
Yes I think it will be better if we make a Separate section named Barnstars just beside the "Liked" section. It can contain The barnstars listed serially with their awarding information. Something in this format.

|  Barnstar          | Info                | Notes awaded for                               |
| -------------------- | ----------------- | ------------------------------------------------- |
| Barnstar image | Barnstar info | List nodes where barnstar is awaded |

**Comment by carolineh101** [comment]
I have created a tab for the barnstars:

![screen shot 2016-08-19 at 11 15 34 pm](https://cloud.githubusercontent.com/assets/17236117/17829425/f075adb8-6662-11e6-828c-c7848893414f.png)

However, I'm not sure how to test my code – the barnstars wiki page doesn't seem to be available in development, so I can't add barnstars to the profile.

**Comment by ananyo2012** [comment]
You can just download the barnstar images and use it for work when offline. When online you can just use the libk of the images in image src. But before that add some dummy data by creating some notes and award barnstars to the users. You can Award barnstars to a user in a note in the dropdown at the top left side of the note beside the like button. Here is a snap
![barnstar](https://cloud.githubusercontent.com/assets/10486343/17829481/cead2d4e-66cd-11e6-8aa0-beba76bd3d52.png)

**Comment by carolineh101** [comment]
Ah, I see now.  Thanks!

**Comment by carolineh101** [comment]
Seems to work!  This is the current formatting I have:

![screen shot 2016-08-19 at 11 37 42 pm](https://cloud.githubusercontent.com/assets/17236117/17829521/07451fda-6666-11e6-9531-6d11ecd946d9.png)

**Comment by ananyo2012** [comment]
Yes Now listing the barnstars in tabular form as I mentioned. One thing you have to keep in mind that a user can be awarded the same barnstar in different notes and by different people. The table should list the barnstar image only once and the instances where they were should go in the last column with a text similar to "by [user] for their worked published on [note title]"

**Comment by jywarren** [comment]
Hi, all -- just a note but maybe you figured this out already -- the formatting you showed looks perfect, @carolineh101, and the larger custom-graphics barnstars are an older system you don't have to worry about right now -- you shouldn't need any custom images (they're only for the older system). 

So, no need for these right now, that's a separate issue (#265) for those older barnstars: 

![screenshot 2016-08-20 at 9 35 02 am](https://cloud.githubusercontent.com/assets/24359/17831540/7316db1a-66b9-11e6-86f1-851410736302.png)

The table as shown looks perfect!

**Comment by ananyo2012** [comment]
Ok then we can just go with the Barnstar name. What do you say?

**Comment by jywarren** [comment]
Sorry, Ananyo, where do you mean? The newer system still has custom names, just not custom images. But maybe I misunderstand?

**Comment by ananyo2012** [comment]
I mean instead og using the Barnstar images we should just show the Barnstar name in the table

**Comment by jywarren** [comment]
the table as it's shown right now in Caroline's screenshot looks perfect.
The yellow "star" icon is all we need -- if this is a pic of the code, then
it looks ready for a PR!

**Comment by ananyo2012** [comment]
I have edited the table. Now it should be clear. Though I love the Barnstar icons! They are great! In fact I was thinking of putting a Barnstar icon of the Basic Barnstar which was awarded to me by Stevie in my Profile Page! :blush:

**Comment by ananyo2012** [comment]
Ok you want just that? Actually I was thinking of consolidating it as I showed in the Table I made above. My idea is showing the Barnstar name only once and list all the instances that were awarded for the same Barnstar.

**Comment by jywarren** [comment]
Yes, the custom barnstars are super nice, but we haven't figured out yet how to integrate them into the system. But we're not getting rid of them -- #265 is devoted to figuring that out.

**Comment by jywarren** [comment]
I'm fine with either a list or a table, I leave it up to Caroline. Thanks!

**Comment by carolineh101** [comment]
I think I'm going to go ahead and submit a PR!
